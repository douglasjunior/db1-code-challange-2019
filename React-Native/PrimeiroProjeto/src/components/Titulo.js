import React from 'react';
import { Text } from 'react-native';

const Titulo = ({ textColor, children }) => {
    const style = {
        color: textColor,
        backgroundColor: 'yellow'
    };
    return (
        <Text style={style}>{children}</Text>
    );
};

export default Titulo;
