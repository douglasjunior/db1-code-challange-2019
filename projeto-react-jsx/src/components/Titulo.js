import React from 'react';

const Titulo = (props) => {
    const style = {
        color: props.textColor,
        backgroundColor: 'yellow'
    }

    return (
        <h1 style={style}>
          {props.children}
        </h1>
    );
};

export default Titulo;
