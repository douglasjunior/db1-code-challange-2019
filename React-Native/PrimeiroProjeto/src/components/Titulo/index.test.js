/**
 * @format
 */

import { Text } from 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Titulo from './';

describe('Snapshot tests', () => {

  test('Título snapshot', () => {
    const treeJSON = renderer
      .create(<Titulo>Meu título</Titulo>)
      .toJSON();
    expect(treeJSON).toMatchSnapshot();
  });

});

describe('Shallow tests', () => {

  test('Test color style', () => {
    const comp = shallow(
      <Titulo textColor='green'>
        Meu título verde
      </Titulo>
    );
    expect(comp.getElement().props.style.color).toBe('green');
  });

  test('Test element type', () => {
    const comp = shallow(
      <Titulo textColor='green'>
        Meu título verde
      </Titulo>
    );
    expect(comp.getElement().type).toBe(Text);
  });

  test('Test title text', () => {
    const comp = shallow(
      <Titulo textColor='green'>
        Meu título verde
      </Titulo>
    );
    expect(comp.getElement().props.children).toBe('Meu título verde');
  });

});
