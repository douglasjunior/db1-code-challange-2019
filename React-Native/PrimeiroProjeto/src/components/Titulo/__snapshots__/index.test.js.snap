// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Título snapshot 1`] = `
<Text
  style={
    Object {
      "backgroundColor": "yellow",
      "color": undefined,
    }
  }
>
  Meu título
</Text>
`;
